pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -699,6 +705,7 @@ func (m *kubeGenericRuntimeManager) restoreSpecsFromContainerLabels(ctx context.
 			Name:                       l.PodName,
 			Namespace:                  l.PodNamespace,
 			DeletionGracePeriodSeconds: a.PodDeletionGracePeriod,
+			Annotations:                annotations,
```

Comment:
while comparing the patches from 1.28:

https://github.com/datadog/kubernetes/compare/506050d61cf291218dfbd41ac93913945c9aa0da...release-1.28-dd-v1.28.5-dd.1#diff-d86ffc24d8e4c9deafd3e39e8f4ee3546ee46a31dcea8ea6526443b3f4bffa28R716

i noticed that we're missing a block of code under this

```go
			// after a kubelet restart, it's not 100% certain that the
			// pod we're given has the container we need in the spec
			// -- we try to recover that here.
			restoredPod, restoredContainer, err := m.restoreSpecsFromContainerLabels(ctx, containerID)
			...
```

is it no longer needed?

Reply:
Brought it back because folk at lyft still have it, but imho we don't necessarily need it.
It's not upstream, and there is no reason for the container spec to be missing if the pod is not nil.